cachi2/core/package_managers/gomod.py

```diff
@@ -1012,26 +1029,17 @@ def _resolve_gomod(
         go_work, go, go_list, run_params, app_dir, version_resolver
     )
 
-    def go_list_deps(pattern: Literal["./...", "all"]) -> Iterator[ParsedPackage]:
-        """Run go list -deps -json and return the parsed list of packages.
-
-        The "./..." pattern returns the list of packages compiled into the final binary.
-
-        The "all" pattern includes dependencies needed only for tests. Use it to get a more
-        complete module list (roughly matching the list of downloaded modules).
-        """
-        cmd = [*go_list, "-deps", "-json=ImportPath,Module,Standard,Deps", pattern]
-        return map(ParsedPackage.model_validate, load_json_stream(go(cmd, run_params)))
-
     package_modules = [
-        module for pkg in go_list_deps("all") if (module := pkg.module) and not module.main
+        module
+        for pkg in _go_list_deps(go, "all", run_params)
+        if (module := pkg.module) and not module.main
```

Comment:
Nitpick for the sake readability. What do you think of something like:
```python
not_main = lambda pkg: (module := pkg.module) and not module.main
package_modules = [pkg.module for pkg in _go_list_deps(go, "all", run_params) if not_main(pkg)]
```
It is a multiline one-liner otherwise.

Reply:
lambda - why didn't I think of it?! I hate multiline comprehensions. Thanks!

Reply:
lambda would not fly -> https://www.flake8rules.com/rules/E731.html so I solved it in a different way and still making this a one-liner.